qiskit/circuit/random/utils.py

```diff
@@ -11,15 +11,415 @@
+
+                    for current_gate, num_gate_params, qubit_idx in zip(
+                        extra_1q_gates["class"], extra_1q_gates["num_params"], qubit_idx_not_used
+                    ):
+
+                        param = parameters_1q[:num_gate_params]
+                        parameters_1q = parameters_1q[num_gate_params:]
+                        qc._append(
+                            CircuitInstruction(
+                                operation=current_gate(*param),
+                                qubits=[qubits[qubit_idx]],
+                            )
+                        )
+
+        else:
+            for current_gate, num_gate_params, edge in zip(
+                gate_choices["class"],
+                gate_choices["num_params"],
+                edge_choices,
+            ):
+                params = parameters[:num_gate_params]
+                parameters = parameters[num_gate_params:]
+                operation = current_gate(*params)
+                control_qubit, target_qubit = tuple(edge)
+
+                if control_qubit in qubit_idx_not_used and target_qubit in qubit_idx_not_used:
+                    qc._append(
+                        CircuitInstruction(
+                            operation=operation,
+                            qubits=[qubits[control_qubit], qubits[target_qubit]],
+                        )
+                    )
+                    qubit_idx_used.update(set(edge))
+                    qubit_idx_not_used = qubit_idx_not_used - qubit_idx_used
+                    edges_used[(control_qubit, target_qubit)] += 1
+
+            if insert_1q_oper:
+
+                if not len(qubit_idx_not_used) == 0:
+                    # Some extra 1Q Gate in to fill empty qubits for this while iteration.
+
+                    extra_1q_gates = rng.choice(
+                        gates_1q, size=len(qubit_idx_not_used), replace=True
+                    )
+                    cumsum_params = np.cumsum(extra_1q_gates["num_params"], dtype=np.int64)
+                    parameters_1q = rng.uniform(0, 2 * np.pi, size=cumsum_params[-1])
+
+                    for current_gate, num_gate_params, qubit_idx in zip(
+                        extra_1q_gates["class"], extra_1q_gates["num_params"], qubit_idx_not_used
+                    ):
+
+                        param = parameters_1q[:num_gate_params]
+                        parameters_1q = parameters_1q[num_gate_params:]
+                        qc._append(
+                            CircuitInstruction(
+                                operation=current_gate(*param),
+                                qubits=[qubits[qubit_idx]],
+                            )
+                        )
+
+        layer_idx += 1
+
+        # check if every edge has been used at-least `min_2q_gate_per_edge` number of times.
+        reached_depth = np.array(list(edges_used.values())) >= min_2q_gate_per_edge
```

Comment:
I wanted to achieve a do-while loop behavior,
There is no need to check the condition for the first time, because its certain that all the qubits would be idle.
That's why I had checked the condition at the very last, and to make the code look pretty used a variable `stop`.

Anyways, changed the code to your liking :)